oracle/oracles/pd.go

```diff
@@ -241,28 +344,145 @@ func (o *pdOracle) getLastTSWithArrivalTS(txnScope string) (*lastTSO, bool) {
 	return last, true
 }
 
+// nextUpdateInterval calculates the next interval to update the low resolution ts. When this is triggered by timer
+// tick, requiredStaleness should be 0. This function may also be called in case there is a stale read / snapshot read
+// requesting a short staleness (i.e., the read operation specifies the version it wants to read instead of allocating
+// from PD, and the required version is very close to the current time). In this case, the requested staleness should be
+// passed to this function as the requiredStaleness parameter.
+func (o *pdOracle) nextUpdateInterval(now time.Time, requiredStaleness time.Duration) time.Duration {
+	o.adaptiveUpdateIntervalState.mu.Lock()
+	defer o.adaptiveUpdateIntervalState.mu.Unlock()
+
+	configuredInterval := time.Duration(o.lastTSUpdateInterval.Load())
+	currentAdaptiveInterval := time.Duration(o.adaptiveLastTSUpdateInterval.Load())
+	if configuredInterval <= minAllowedAdaptiveUpdateTSInterval {
```

Comment:
If `configuredInterval <= minAllowedAdaptiveUpdateTSInterval` is true, then `adaptiveLastTSUpdateInterval` can be set to `configuredInterval`. The variable `minAllowedAdaptiveUpdateTSInterval` doesn't behave like its name.

How about defining a `minUpdateTSInterval` and both `lastTSUpdateInterval` and `adaptiveLastTSUpdateInterval` should less than `minUpdateTSInterval`.

Reply:
The name `minAllowedAdaptiveUpdateTSInterval` expected to limit the *adaptive* update ts interval, to avoid it automatically performs the update too frequently. If the user configures a short interval intentionally, we adopt the user's choice.

Reply:
By the way I tried to refine the code of the nextUpdateInterval function, which extracted each branches into small functions and unified the log printing. Please see if it's better than the previous one.